e2e/reset-modal.spec.ts

```diff
@@ -44,21 +44,34 @@ test('should render the modal content correctly', async ({ page }) => {
 });
 
 test('User can reset challenge', async ({ page }) => {
+  const initialText = 'CatPhotoApp';
+  const initialFrame = page
+    .frameLocator('iframe[title="challenge preview"]')
+    .getByText(initialText);
+
+  const updatedText = 'Only Dogs';
+  const updatedFrame = page
+    .frameLocator('iframe[title="challenge preview"]')
+    .getByText(updatedText);
+
   await page.goto(
     '/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-2'
   );
 
+  // Building the preview can take a while
+  await expect(initialFrame).toBeVisible({ timeout: 10000 });
+
   const editorPaneLabel =
     'Editor content;Press Alt+F1 for Accessibility Options.';
-  const challengeSolution = '<h2>Cat Photos</h2>';
 
-  await page.getByLabel(editorPaneLabel).fill(challengeSolution);
-  await expect(
-    page
-      .frameLocator('iframe[title="challenge preview"]')
-      .getByRole('heading', { name: 'Cat Photos' })
-  ).toBeVisible();
+  // Modify the text in the editor pane, clearing first, otherwise the existing
+  // text will be selected before typing
+  await page.getByLabel(editorPaneLabel).fill('');
+  await page.getByLabel(editorPaneLabel).fill(updatedText);
+  await expect(updatedFrame).toBeVisible();
```

Comment:
```suggestion
  await expect(updatedFrame).toBeVisible({ timeout: 10000});
```

We should maybe put a wait in here too - sometimes the preview reload can take a minute.

Reply:
Can't hurt.